turbo/rpchelper/helper.go

```diff
@@ -34,28 +33,36 @@ func GetCanonicalBlockNumber(blockNrOrHash rpc.BlockNumberOrHash, tx kv.Tx, filt
 
 func _GetBlockNumber(requireCanonical bool, blockNrOrHash rpc.BlockNumberOrHash, tx kv.Tx, filters *Filters) (blockNumber uint64, hash common.Hash, latest bool, err error) {
 	var latestBlockNumber uint64
-	if latestBlockNumber, err = stages.GetStageProgress(tx, stages.Execution); err != nil {
+	if latestBlockNumber, err = GetLatestBlockNumber(tx); err != nil {
```

Comment:
Ok, you actually need both, because previous `latestBlockNumber` is still needed to return `latest` value correctly - this determines whether we can use PlainState and state cache

Reply:
Well if we use the previous then we don't favour to use ForkchoiceHeadHash.